Enable early-in-memory-session-create spec on WOA for 10-x-y

This started failing in the Chromium roll for 10-x-y #24706,

eg on https://github.visualstudio.com/electron/_build/results?buildId=78029&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=1824:

```
Unhandled exception in main spec runner: AssertionError: should have exited with code 0: expected 3221225477 to equal 0
    at ChildProcess.<anonymous> (C:\vsts-agent-win-x86-2.166.4\_work\1\s\src\electron\spec-main\crash-spec.ts:26:23)
    at ChildProcess.emit (events.js:310:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12) {
  showDiff: true,
  actual: 3221225477,
  expected: 0
}
not ok 1459 crash cases the "early-in-memory-session-create" case should not crash
  AssertionError: should have exited with code 0: expected 3221225477 to equal 0
      at ChildProcess.<anonymous> (electron\spec-main\crash-spec.ts:26:23)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
not ok 1460 crash cases "after each" hook for "the "early-in-memory-session-create" case should not crash"
  AssertionError: all child processes should have exited cleanly: expected [ Array(1) ] to have a length of 0 but got 1
      at Context.<anonymous> (electron\spec-main\crash-spec.ts:38:30)
      at processImmediate (internal/timers.js:456:21)
[5240:0810/110300.660:WARNING:discardable_shared_memory_manager.cc(432)] Some MojoDiscardableSharedMemoryManagerImpls are still alive. They will be leaked.
Received fatal exception EXCEPTION_ACCESS_VIOLATION
Backtrace:
	v8::Unlocker::~Unlocker [0x00007FF7E2ABADD0+26944]
	v8::MemorySpan<char const >::MemorySpan<char const > [0x00007FF7E1AC2C4C+2588]
	std::__1::__vector_base<v8::CpuProfileDeoptInfo,std::__1::allocator<v8::CpuProfileDeoptInfo> >::clear [0x00007FF7E197A4DC+26524]
	cppgc::Platform::GetForegroundTaskRunner [0x00007FF7E1A82ECC+12204]
	v8::ResourceConstraints::max_zone_pool_size [0x00007FF7E1AF5F18+41848]
	v8::ResourceConstraints::max_zone_pool_size [0x00007FF7E1AF5EC8+41768]
	uv_gettimeofday [0x00007FF7E531E19C+341532]
	uv_gettimeofday [0x00007FF7E531E39C+342044]
	uv_mutex_unlock [0x00007FF7E49B1B18+2468232]
	uv_mutex_unlock [0x00007FF7E49AC9F0+2447456]
	uv_mutex_unlock [0x00007FF7E49A3674+2409700]
	uv_mutex_unlock [0x00007FF7E49AACA0+2439952]
	uv_fs_get_path [0x00007FF7E4151C28+5838872]
	uv_fs_get_path [0x00007FF7E4151D2C+5839132]
	uv_fs_get_path [0x00007FF7E4151210+5836288]
	uv_fs_get_path [0x00007FF7E4150F48+5835576]
	uv_fs_get_path [0x00007FF7E41510E8+5835992]
	uv_fs_get_path [0x00007FF7E4150904+5833972]
	uv_fs_get_path [0x00007FF7E3C903A0+851856]
	IsSandboxedProcess [0x00007FF7E4FAE77C+463484]
	v8::CpuProfilingOptions::max_samples [0x00007FF7E27496B0+936912]
	Ordinal0 [0x00007FF7E19613CC+5068]
	uv_random [0x00007FF7E89FA9A4+18610468]
	uv_random [0x00007FF7E89FAA30+18610608]
	BaseThreadInitThunk [0x00007FFE2B994974+52]
	RtlUserThreadStart [0x00007FFE2D6F86F4+68]
✗ Electron tests failed with code 0xc0000005.
````

--- a/azure-pipelines-woa.yml
+++ b/azure-pipelines-woa.yml
@@ -87,7 +87,11 @@ steps:
 
 - powershell: |
     Get-Process | Where Name –Like "electron*" | Stop-Process
-    Get-Process | Where Name –Like "MicrosoftEdge*" | Stop-Process
     Get-Process | Where Name –Like "msedge*" | Stop-Process
   displayName: 'Kill processes left running from last test run'
+  condition: always()
+
+- powershell: |
+    Remove-Item -path $env:APPDATA/Electron* -Recurse
+  displayName: 'Delete user app data directories'
   condition: always()

--- a/spec-main/crash-spec.ts
+++ b/spec-main/crash-spec.ts
@@ -2,7 +2,6 @@ import { expect } from 'chai';
 import * as cp from 'child_process';
 import * as fs from 'fs';
 import * as path from 'path';
-import { ifdescribe } from './spec-helpers';
 
 const fixturePath = path.resolve(__dirname, 'fixtures', 'crash-cases');
 
@@ -31,8 +30,7 @@ const runFixtureAndEnsureCleanExit = (args: string[]) => {
   });
 };
 
-// TODO re-enable this test for WOA
-ifdescribe(process.platform !== 'win32' || process.arch !== 'arm64')('crash cases', () => {
+describe('crash cases', () => {
   afterEach(() => {
     for (const child of children) {
       child.kill();

--- a/spec-main/fixtures/crash-cases/early-in-memory-session-create/index.js
+++ b/spec-main/fixtures/crash-cases/early-in-memory-session-create/index.js
@@ -2,5 +2,7 @@ const { app, session } = require('electron');
 
 app.on('ready', () => {
   session.fromPartition('in-memory');
-  process.exit(0);
+  setImmediate(() => {
+    process.exit(0);
+  });
 });

